huntsman/observatory.py

```diff
@@ -316,6 +314,105 @@ def take_flat_fields(self, camera_names=None, alt=None, az=None,
+                         ):
+        """Take dark fields for each camera with all the exposure times
+           in the targets list.
+
+        Args:
+            exptimes_list (list): List of exposure times for darks
+            wait_interval (float, optional): Time in between dark exposures
+            camera_list (list, optional): List of cameras to use for darks
+            n_darks (int, optional): Number of darks to be taken per exptime
+            *args (TYPE): Description
+            **kwargs (TYPE): Description
+
+        """
+
+        if camera_names is None:
+            cameras_all = self.cameras
+        else:
+            cameras_all = {c: self.cameras[c] for c in camera_names}
+
+        self.logger.debug('Using cameras {}'.format(cameras_all))
+
+        image_dir = self.config['directories']['images']
+
+        self.logger.debug("Going to take {} dark fields for each of these exposure times {}".
+                          format(n_darks, exptimes))
+
+        # List to check that the final number of darks is equal to the number
+        # of cameras times the number of exptimes times n_darks.
+        total_exposures = []
+
+        # Loop over cameras.
+        for exptime in exptimes:
+
+            self.status()  # Seems to help with reading coords
+
+            start_time = utils.current_time()
+
+            if isinstance(exptime, u.Quantity):
+                exptime = exptime.to_value(u.second)
+
+            # Loop over exposure times for each camera.
+            for num in range(n_darks):
+
+                self.logger.debug('Darks sequence #{} of exposure time {}s'.
+                                  format(num, exptime))
+
+                camera_events = dict()
+
+                # Take a given number of exposures for each exposure time.
+                for camera in cameras_all.values():
+
+                    self.logger.debug('Camera {} is exposing for {}s'.
+                                      format(camera.uid, exptime))
+
+                    # Create dark observation
+                    dark_obs = self._create_dark_observation(exptime)
+                    fits_headers = self.get_standard_headers(observation=dark_obs)
+                    # Common start time for cameras
+                    fits_headers['start_time'] = utils.flatten_time(start_time)
+
+                    filename = '{}/darks/{}/{}/{}.{}'.format(
+                        image_dir,
+                        camera.uid,
+                        dark_obs.seq_time,
+                        'dark_{:02d}_{}s'.format(num, int(exptime)),
+                        camera.file_extension)
+
+                    # Take picture and get event
+                    camera_event = camera.take_observation(
+                        dark_obs,
+                        fits_headers,
+                        filename=filename,
+                        exptime=exptime,
+                        dark=True,
+                        blocking=True
```

Comment:
I don't think you want to block here as it waits below.

```suggestion
                        blocking=False
```